<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>fit_grains_results_dialog</class>
 <widget class="QDialog" name="fit_grains_results_dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Fit Grains Results</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <widget class="QTableView" name="table_view">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>150</height>
       </size>
      </property>
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
     </widget>
     <widget class="QWidget" name="plot_widget" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="dialog_controls" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="export_button">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Export</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QGroupBox" name="plot_controls">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="title">
             <string>Options</string>
            </property>
            <layout class="QFormLayout" name="formLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="projection_label">
               <property name="text">
                <string>Projection:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="projection">
               <item>
                <property name="text">
                 <string>Orthographic</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Perspective</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="plot_color_label">
               <property name="text">
                <string>Color By:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="plot_color_option">
               <item>
                <property name="text">
                 <string>Completeness</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Goodness of Fit</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>XX Strain</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>YY Strain</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>ZZ Strain</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>YZ Strain</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>XZ Strain</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>XY Strain</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="hide_axes">
               <property name="text">
                <string>Hide Axes</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QToolButton" name="set_view_direction">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Reset Direction</string>
               </property>
               <property name="popupMode">
                <enum>QToolButton::InstantPopup</enum>
               </property>
               <property name="toolButtonStyle">
                <enum>Qt::ToolButtonTextOnly</enum>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="QCheckBox" name="convert_strain_to_stress">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Convert strain to stress. Requires a material where the elastic stiffness tensor is defined (see the properties tab in the materials panel).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>Convert Strain to Stress</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0" colspan="2">
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLabel" name="glyph_size_label">
                 <property name="text">
                  <string>Glyph Size</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSlider" name="glyph_size_slider">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>2000</number>
                 </property>
                 <property name="singleStep">
                  <number>1</number>
                 </property>
                 <property name="pageStep">
                  <number>2</number>
                 </property>
                 <property name="value">
                  <number>1</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="reset_glyph_size">
                 <property name="text">
                  <string>Reset</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="color_map_label">
               <property name="text">
                <string>Color Map</string>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QComboBox" name="color_maps">
               <property name="styleSheet">
                <string notr="true">combobox-popup: 0;</string>
               </property>
               <property name="editable">
                <bool>false</bool>
               </property>
               <property name="maxVisibleItems">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="8" column="0" colspan="2">
              <widget class="QGroupBox" name="ranges_group">
               <property name="title">
                <string>Ranges</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_2">
                <item row="1" column="2">
                 <widget class="ScientificDoubleSpinBox" name="range_y_0">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="2" column="4">
                 <widget class="ScientificDoubleSpinBox" name="range_z_1">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="ScientificDoubleSpinBox" name="range_z_0">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="4">
                 <widget class="ScientificDoubleSpinBox" name="range_y_1">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="3">
                 <widget class="QLabel" name="y_range_separator">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>-</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="2" column="3">
                 <widget class="QLabel" name="z_range_separator">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>-</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="ScientificDoubleSpinBox" name="range_x_0">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QLabel" name="x_range_label">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>X:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="4">
                 <widget class="ScientificDoubleSpinBox" name="range_x_1">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="keyboardTracking">
                   <bool>false</bool>
                  </property>
                  <property name="singleStep">
                   <number>0</number>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="y_range_label">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Y:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="3">
                 <widget class="QLabel" name="x_range_separator">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>-</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignCenter</set>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="z_range_label">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Z:</string>
                  </property>
                 </widget>
                </item>
                <item row="3" column="0" colspan="5">
                 <widget class="QPushButton" name="reset_ranges">
                  <property name="text">
                   <string>Reset Ranges</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item row="5" column="0" colspan="2">
              <widget class="QPushButton" name="export_stresses">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Export stresses to a compressed npz file as Mandel-Voigt vectors. The data is saved in the &amp;quot;stresses&amp;quot; key as an (n, 6) shape, where `n` is the number of grains.&lt;/p&gt;&lt;p&gt;&amp;quot;Convert Strain to Stress&amp;quot; must be enabled to use this.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>Export Stresses</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QCheckBox" name="depth_shading">
               <property name="text">
                <string>Depth Shading</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QCheckBox" name="cylindrical_reference">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;For the 3D coordinates, use a cylindrical reference frame (ρ, φ, Y) instead of cartesian coordinates.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>Cylindrical  Reference</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="canvas_layout"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="toolbar_layout"/>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="button_box">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ScientificDoubleSpinBox</class>
   <extends>QSpinBox</extends>
   <header>hexrd/ui/scientificspinbox</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>table_view</tabstop>
  <tabstop>export_button</tabstop>
  <tabstop>projection</tabstop>
  <tabstop>plot_color_option</tabstop>
  <tabstop>hide_axes</tabstop>
  <tabstop>set_view_direction</tabstop>
  <tabstop>depth_shading</tabstop>
  <tabstop>cylindrical_reference</tabstop>
  <tabstop>convert_strain_to_stress</tabstop>
  <tabstop>export_stresses</tabstop>
  <tabstop>glyph_size_slider</tabstop>
  <tabstop>reset_glyph_size</tabstop>
  <tabstop>color_maps</tabstop>
  <tabstop>range_x_0</tabstop>
  <tabstop>range_x_1</tabstop>
  <tabstop>range_y_0</tabstop>
  <tabstop>range_y_1</tabstop>
  <tabstop>range_z_0</tabstop>
  <tabstop>range_z_1</tabstop>
  <tabstop>reset_ranges</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>fit_grains_results_dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>fit_grains_results_dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>convert_strain_to_stress</sender>
   <signal>toggled(bool)</signal>
   <receiver>export_stresses</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>462</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
